File: Tutorials/Tuckshop/Client/Tuckshop.WebApp/src/Domain/Models/Category.ts
import { ModelBase, Rules, Validation } from '@singularsystems/neo-core';

export default class Category extends ModelBase {
    static typeName = "Category";

    constructor() {
        super();
        this.makeObservable();
    }

    public categoryId: number = 0;

    @Rules.Required()
    @Rules.StringLength(100)
    public categoryName: string = "";

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<Category>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        if (this.isNew || !this.categoryName) {
            return "New category";
        } else {
            return this.categoryName;
        }
    }
}

File: Tutorials/Tuckshop/Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Category.cs
namespace Tuckshop.Core.Models
{
  using System.ComponentModel.DataAnnotations;
  using Neo.Model;

  /// <summary>
  /// Category class to represent a category in the system.
  /// </summary>
  public class Category : ModelBase<Category>
  {
    /// <summary>
    /// Gets or sets the Category Id.
    /// </summary>
    public int CategoryId { get; set; }

    /// <summary>
    /// Gets or sets the category name.
    /// </summary>
    [Required(AllowEmptyStrings = false)]
    [StringLength(100)]
    public string CategoryName { get; set; } = string.Empty;

  }
}
